### PR TITLE
added ansible_password, fixed sasl, scram256 users, ssl/,tls more exp…

### PR DESCRIPTION
**Pull request by thecrazymonkey**
…licit

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)
Added propagation of ansible_password - my customer uses user/password for the ssh login.
For custom listeners 
- fixed sasl_protocol not being returned  
- fixed scram256 user not being returned
- changed returned scram_sha_512_users to be returned  in scram_users as cp-ansible does not have 512_users (scram_users are 512_users)
- added more explicit setting of tls/mTLS in the listener spec, same for SASL

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Yes

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Tested on deployment with TLS/mTLS listeners, TOKEN listener, SASL_PLAIN, SASL_SCRAM 

# Checklist:

- [ ] Any variable/code changes have been validated to be backwards compatible (doesn't break upgrade)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If required, I have ensured the changes can be discovered by cp-ansible discovery codebase
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: discovery/service/kafka_broker.py [hunk 1]
```diff
@@ -106,10 +106,12 @@ def __get_user_dict(self, service_prop: dict, key: str) -> dict:
         # parse line to get the user configs
         for token in jaas_config.split():
             if "=" in token:
-                username, password = token.split('=')
-                if username.startswith('user_'):
-                    principal = username.replace('user_', '')
-                    # Sanitize the password
+                key, value = token.split('=')
+                if key.startswith('username'):
```
Comment (nsharma-git) [inline discovery/service/kafka_broker.py:110]: I can't recall why it was `user_`. Can you please attach some samples in support of this change. I'm not sure if this is non breaking change.

File: discovery/service/kafka_broker.py [hunk 2]
```diff
@@ -325,25 +327,28 @@ def _build_custom_listeners(self, service_prop: dict) -> tuple:
...
                 _dict.update(self.__get_user_dict(service_prop, key))
                 self.mapped_service_properties.add(key)
+            else:
+                custom_listeners[name]['sasl_protocol'] = 'none'
 
         return self.group, {
             "kafka_broker_custom_listeners": custom_listeners,
-            "sasl_scram_users": default_scram_users,
-            "sasl_scram256_users": default_scram256_users,
-            "sasl_scram512_users": default_scram_sha_512_users,
```
Comment (nsharma-git) [inline discovery/service/kafka_broker.py:346]: Can you please explain the reason to remove scram512 users dict?

File: discovery/manager/manager.py [hunk 3]
```diff
@@ -32,6 +32,7 @@ def my_event_handler(self, event):
     def get_ansible_vars_from_input_context(input_context: InputContext) -> dict:
         vars = dict()
         vars['ansible_user'] = input_context.ansible_user
+        vars['ansible_password'] = input_context.ansible_password
```
Comment (nsharma-git) [inline discovery/manager/manager.py:35]: I believe this change was in master. Not sure was it's not in 7.5.x.

File: discovery/service/kafka_broker.py [hunk 4]
```diff
@@ -325,25 +327,28 @@ def _build_custom_listeners(self, service_prop: dict) -> tuple:
...
+                logger.info(f"SSL mutual auth enabled for listener {name} {ssl_mutual_enabled}")
+                if ssl_mutual_enabled == 'required':
+                    custom_listeners[name]['ssl_mutual_auth_enabled'] = True
+                else:
+                    custom_listeners[name]['ssl_mutual_auth_enabled'] = False
 
             sasl_protocol = service_prop.get(key1)
             if sasl_protocol is not None:
-                custom_listeners[name]['sasl_protocol']: sasl_protocol
+                custom_listeners[name]['sasl_protocol'] = sasl_protocol
```
Comment (nsharma-git) [inline discovery/service/kafka_broker.py:339]: Thanks!

File: discovery/service/kafka_broker.py (hunk 2 above)
Comment (thecrazymonkey) [reply discovery/service/kafka_broker.py:346]: cp-ansible has only sasl_scram_users or sasl_scram256_users, the former is used for SCRAM-SHA-512 so just wanted this to be more inline with what cp-ansible does.

Comment (nsharma-git) [reply discovery/service/kafka_broker.py:346]: Discovery would be running against the cluster which is mostly probably not maintained by cp-ansible. If such cluster has scram512 users, what would be our strategy. Are we planning to drop them?

File: discovery/service/kafka_broker.py (hunk 1 above)
Comment (nsharma-git) [reply discovery/service/kafka_broker.py:110]: thanks. To make it safer, can we keep both `user_` and `username` for now. Do you see any consequences ?

File: discovery/service/kafka_broker.py (hunk 2 above)
Comment (thecrazymonkey) [reply discovery/service/kafka_broker.py:346]: Not dropping - we get the scram config from the sasl.mechanism which is either SCRAM-SHA-256 or SCRAM-SHA-512. Then - depending on which type it is we fill the users dict. To match cp-ansible we use the scram_users and scram265_users. We are only getting the scram users by parsing the sasl.jaas.config line anyway - the rest is in Zookeeper and we would need to use CLI - kafka-configs to get usernames from it (can't get passwords).

File: discovery/service/kafka_broker.py (hunk 1 above)
Comment (thecrazymonkey) [reply discovery/service/kafka_broker.py:110]: yeah, good point. Let me do bit more testing on this.

Comment (thecrazymonkey) [reply discovery/service/kafka_broker.py:110]: Added back the original code so now it covers both sasl plain form of credentials as well as scram. Tested with both

Comment (thecrazymonkey) [reply discovery/service/kafka_broker.py:110]: Here's a sample sasl plain jaas config:
listener.name.plain.plain.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="admin" password="admin-secret"  user_admin="admin-secret" user_client="client-secret" user_schema_registry="schema_registry-secret" user_kafka_connect="kafka_connect-secret" user_kafka_rest="kafka_rest-secret" user_ksql="ksql-secret" user_control_center="control_center-secret";

**Review by nsharma-git — APPROVED** [review]

**Pull request merged**